src/main/java/xbot/common/command/BaseParallelCommandGroup.java

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+
+package xbot.common.command;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+
+/**
+ * A command composition that runs a set of commands in parallel, ending when the last command ends.
+ *
+ * <p>The rules for command compositions apply: command instances that are passed to it cannot be
+ * added to any other composition or scheduled individually, and the composition requires all
+ * subsystems its components require.
+ *
+ * <p>This class is provided by the NewCommands VendorDep
+ */
+public class BaseParallelCommandGroup extends Command {
+  // maps commands in this composition to whether they are still running
+  // LinkedHashMap guarantees we iterate over commands in the order they were added (Note that
+  // changing the value associated with a command does NOT change the order)
+  private final Map<Command, Boolean> m_commands = new LinkedHashMap<>();
+  private boolean m_runWhenDisabled = true;
+  private InterruptionBehavior m_interruptBehavior = InterruptionBehavior.kCancelIncoming;
+
+  /**
+   * Creates a new ParallelCommandGroup. The given commands will be executed simultaneously. The
+   * command composition will finish when the last command finishes. If the composition is
+   * interrupted, only the commands that are still running will be interrupted.
+   *
+   * @param commands the commands to include in this composition.
+   */
+  @SuppressWarnings("this-escape")
+  public BaseParallelCommandGroup(Command... commands) {
+    addCommands(commands);
+  }
+
+  /**
+   * Adds the given commands to the group.
+   *
+   * @param commands Commands to add to the group.
+   */
+  public final void addCommands(Command... commands) {
+    if (m_commands.containsValue(true)) {
+      throw new IllegalStateException(
+          "Commands cannot be added to a composition while it's running");
+    }
+
+    CommandScheduler.getInstance().registerComposedCommands(commands);
+
+    for (Command command : commands) {
+      if (!Collections.disjoint(command.getRequirements(), getRequirements())) {
+        throw new IllegalArgumentException(
+            "Multiple commands in a parallel composition cannot require the same subsystems");
+      }
+      m_commands.put(command, false);
+      addRequirements(command.getRequirements());
+      m_runWhenDisabled &= command.runsWhenDisabled();
+      if (command.getInterruptionBehavior() == InterruptionBehavior.kCancelSelf) {
+        m_interruptBehavior = InterruptionBehavior.kCancelSelf;
+      }
+    }
+  }
+
+  @Override
+  public final void initialize() {
+    super.initialize();
```

Comment:
The only delta is here calling super and same thing in end(). Jerks had to mark it final ✊🏻